### PR TITLE
Assume 3 when can't determine API version

### PR DESCRIPTION
**Pull request by jhernand**
Currently when using version 4 of the SDK is enabled the provider raises
an exception when it can't determine what versions of the API are
supported by the oVirt server. But this is a normal situation. It
happens in general when the credentials of the server are not known yet,
for example when the oVirt server has been discovered scanning a range
of IP addresses. If an exception is reaised then the discovery process
will fail. To avoid this issue this patch changes the provider so that
it tolerates this situation, and assumes that the server supports at
least version 3 of the API.

This patch addresses the following bug:

  RHV41 Provider Discovery failure
  https://bugzilla.redhat.com/1502963

### CODE REVIEW

**Comment by jhernand** [comment]
@pkliczewski @borod108 please review.

File: app/models/manageiq/providers/redhat/infra_manager/api_integration.rb [hunk 1]
```diff
@@ -85,8 +85,12 @@ def supported_api_versions_from_sdk
...
     probe_results.map(&:version) if probe_results
   rescue => error
-    _log.error("Error while probing supported api versions #{error}")
-    raise
+    # Note that errors when trying to find the supported API versions are perfectly normal, in particular authorization
+    # errors are expected, as in many situations, for example during discovery, the user name and the password aren't
+    # yet known. In these situations we *must* return an empty array, to indicate to the caller that it wasn't possible
+    # to determine the supported API versions.
+    _log.info("Can't determine the API versions supported by the server: #{error}")
+    []
```
Comment (pkliczewski) [inline app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: we won't have information why the probing failed. In situations like wrong credentials we won't state what happened. Am I right?

Comment (jhernand) [reply app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: We won't have information on why determining the supported versions failed. But that is just one part of the process of checking the credentials. After determining the supported versions we send additional requests to check the credentials (if the credentials are available). Those requests will fail, and the error will be reported to the user. The net result is that the user will see a message in the UI saying that the user name or password isn't correct.

Comment (pkliczewski) [reply app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: OK, so if the user gets this information I am fine with the change.

File: app/models/manageiq/providers/redhat/infra_manager/api_integration.rb [hunk 2]
```diff
@@ -226,7 +230,9 @@ def with_version4_vm_service(vm)
   end
 
   def highest_supported_api_version
-    supported_api_versions.sort.last
+    versions = supported_api_versions
+    return '3' if versions.empty?
```
Comment (borod108) [inline app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: Maybe versions.sort.last || '3' would be shortwr

Comment (borod108) [reply app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: I mean supported_api_versions.sort.last || '3'

Comment (jhernand) [reply app/models/manageiq/providers/redhat/infra_manager/api_integration.rb]: Yes, that makes sense, thanks @borod108, I am changing it.

**Comment by jhernand** [comment]
The new version of the pull request addresses the comments from @borod108.

**Pull request merged**